service/header/header.go

```diff
@@ -32,10 +33,10 @@ func (eh *ExtendedHeader) MarshalBinary() ([]byte, error) {
 	return MarshalExtendedHeader(eh)
 }
 
-// MarshalBinary unmarshals ExtendedHeader from binary.
+// UnmarshalBinary unmarshals ExtendedHeader from binary.
 func (eh *ExtendedHeader) UnmarshalBinary(data []byte) error {
 	if eh == nil {
-		return fmt.Errorf("header: cannot UnmarshalBinary - nil ExtendedHeader")
+		return fmt.Errorf("expected: cannot UnmarshalBinary - nil ExtendedHeader")
```

Comment:
The "expected:" seems a bit confusing here. Why not simply:
```suggestion
		return fmt.Errorf("cannot UnmarshalBinary - nil ExtendedHeader")
```